repl/commands.go

```diff
@@ -23,8 +23,10 @@ import (
 var exprSubRe = `(?P<expr>.*)`
 var exprAssignRe = `(\s+=\s+)` + exprSubRe
 var identifier = `[_a-zA-Z][_a-zA-Z0-9]*`
+var typeSuffix = `(\s*:\s*(?P<type_hint>[^=]+))`
 var ws = regexp.MustCompile(`[ \t\r\n]+`)
-var letRe = regexp.MustCompile(`(?P<ident>` + identifier + `)(\s*:\s*(?P<type_hint>\w+))?` + exprAssignRe)
+var letRe = regexp.MustCompile(`(?P<ident>` + identifier + `)` + typeSuffix + `?` + exprAssignRe)
+var declRe = regexp.MustCompile(`(?P<ident>` + identifier + `)` + typeSuffix)
 var delRe = regexp.MustCompile(`(?P<ident>` + identifier + `)`)
```

Comment:
Would it make sense to use `delRe` as a common initial match against the input line? Maybe call it `cmdRe` since you can use it to determine the command name and variable id, then continue matching against the remainder of the line with the other regexes?

Reply:
extracted the prefix part -- I'm going to swap to using antlr before making this anymore complicated in a follow up